acis_thermal_check/state_builder.py

```diff
@@ -216,30 +222,233 @@ def get_validation_states(self, datestart, datestop):
-    since this database is only updated after load products
-    are approved.
-    """
-    def get_prediction_states(self, tlm):
+    def __init__(self, interrupt=False, backstop_file=None, nlet_file = None, cmd_states_db="sybase", logger=None):
         """
-        Get the states used for the thermal prediction.
-        NOT IMPLEMENTED for HDF5StateBuilder.
+        Give the ACISStateBuilder arguments that were passed in 
+        from the command line, and set up the connection to the 
+        commanded states database. 
 
         Parameters
         ----------
-        tlm : dictionary
-            Dictionary containg temperature and other telemetry
+        interrupt : boolean
+            If True, this is an interrupt load.
+        backstop_file : string
+            Path to the backstop file. If a directory, the backstop 
+            file will be searched for within this directory.
+        nlet_file: full path to the Non-Load Event Tracking file
+        cmd_states_db : string
+            Commanded states database server (sybase|acisite). Default:
+            sybase
+        logger : Logger object, optional
+            The Python Logger object to be used when logging.
+        """
+        # Import the BackstopHistory class
+        from backstop_history import BackstopHistory
+
+        # Set the path to the Non-Load Event Tracking Events file
+#        self.nfp = '/data/acis/LoadReviews/NonLoadTrackedEvents.txt'
+
+        # Capture the full path to the NLET file to be used
+        self.nlet_file = nlet_file
+
+        # Create an instance of the Backstop Command class
+        self.BSC = BackstopHistory.BackstopHistory('ACIS-Continuity.txt', self.nlet_file)
+
+        # The Review Load backstop name
+        self.rev_bs_name = None
+        # Normally I would have created the self.rev_bs_cmds attribute
+        # and used that however to work with ATC I changed it to bs_cmds.
+
+        super(ACISStateBuilder, self).__init__(logger=logger)
+        self.interrupt = interrupt
+        self.backstop_file = backstop_file
+
+        # if the user supplied a full path to the backstop file then 
+        # capture the backstop file name and the commands within the backstop file.
+        if backstop_file is not None:
+            # Get tstart, tstop, commands from backstop file in args.oflsdir
+            # These are the REVIEW backstop commands.
+            # NOTE: This method takes every command. it does not eliminate
+            # commands that are not of interest as sot/cmd_states/get_cmds does.
+            rev_bs_cmds,  self.rev_bs_name = self.BSC.get_bs_cmds(self.backstop_file)
+
+        # Store the Review Load backstop commands in the class attribute and
+        # also capture the Review load time of first command (TOFC) and
+        # Time of Last Command (TOLC).
+        self.bs_cmds = rev_bs_cmds
+        self.tstart = rev_bs_cmds[0]['time']
+        self.tstop = rev_bs_cmds[-1]['time']
+
```

Comment:
It looks like now we should try moving the setup of the database into the `StateBuilder` `__init__` method.